### PR TITLE
Update fip-0059.md

### PR DESCRIPTION
**Pull request by irenegia**
- added more details in the actor changes section 
- added more content to the design rationale section
- minor changes to the security consideration section

### CODE REVIEW

File: FIPS/fip-0059.md [hunk 1]
```diff
@@ -53,27 +53,36 @@ Differences between currently deployed PoRep and Synthetic PoRep are limited to
...
-- No proving overhead on the StorageProvider side
-- No impact on PoRep security
+- No need of new Trusted Setup,
+- No proving overhead on the StorageProvider side,
+- No impact on PoRep security.
+
+Note that in the design of Synthetic PoRep we considered the case of increasing the number of PoRep challanges (`N_verified`) in order to be able to have higher storage saving thanks to a reduced number of synthetic challgenes (`N_syn` smaller). However since this requires a new trusted setup and more proving, we decide to go with the proposed parameters that do not change the number of PoRep challanges and still allows significat saving.
+
+An alternative design we considered to reduce the storage between PreCommit and ProveCommit is NI-PoRep (see for example discussion https://github.com/filecoin-project/FIPs/discussions/603 ). NI-PoRep allows to get a 100% storage cost savings and has other benefits (for example, removing the need of PreCommitDeposit) but has a ~8.5x proving overahed that we consider proibitve for now. 
+Not that implementinc Synthetich PoRep now does not prevent us to implement NI-PoRep in the future, once we have solved/reduced theroving overhead problem.
```
Comment (anorth) [inline FIPS/fip-0059.md:84]: A couple of typos on this line.

**Review by anorth — APPROVED** [review]

File: FIPS/fip-0059.md (hunk 1 above)
Comment (luckyparadise) [reply FIPS/fip-0059.md:84]: Hi @irenegia, there is another PR opened for this FIP draft can you confirm that both are ready to be reviewed by editors? I am not sure which one is the latest PR. PS: Alex highlights there are typos above, please confirm they have been fixed. Thank you

Comment (irenegia) [reply FIPS/fip-0059.md]: Yes, both ready now (I have just committed to the typo fixes requested)
@luckyparadise

File: FIPS/fip-0059.md [hunk 2]
```diff
@@ -53,27 +53,36 @@ Differences between currently deployed PoRep and Synthetic PoRep are limited to
     - `RegisteredSealProof_SynthStackedDrg32GiBV1`
     - `RegisteredSealProof_SynthStackedDrg64GiBV1`
 - The allowable delay of the new proof types is the same as the StackedDRG proof types V1.1;
-- No changes in the PreCommit and ProveCommit methods used today
+- No changes in `PreCommitSector` and `ProveCommitSector` methods;
+- `ProveCommitSector` passes the new prove type to the proof verification syscall;
```
Comment (arajasek) [inline FIPS/fip-0059.md:57]:
```suggestion
- `ProveCommitSector` passes the new proof type to the proof verification syscall;
```

File: FIPS/fip-0059.md [hunk 3]
```diff
@@ -53,27 +53,36 @@ Differences between currently deployed PoRep and Synthetic PoRep are limited to
...
-- No need of new Trusted Setup
-- No proving overhead on the StorageProvider side
-- No impact on PoRep security
+- No need of new Trusted Setup,
+- No proving overhead on the StorageProvider side,
+- No impact on PoRep security.
+
+Note that in the design of Synthetic PoRep we considered the case of increasing the number of PoRep challenges (`N_verified`) in order to be able to have higher storage saving thanks to a reduced number of synthetic challenges (`N_syn` smaller). However since this requires a new trusted setup and more proving, we decided to go with the proposed parameters that do not change the number of PoRep challenges and still allows significant saving.
+
+An alternative design we considered to reduce the storage between PreCommit and ProveCommit is NI-PoRep (see for example discussion https://github.com/filecoin-project/FIPs/discussions/603 ). NI-PoRep allows to get a 100% storage cost saving and has other benefits (for example, removing the need of PreCommitDeposit) but has a ~8.5x proving overahed that we consider prohibitive for now.
```
Comment (arajasek) [inline FIPS/fip-0059.md:83]:
```suggestion
An alternative design we considered to reduce the storage between PreCommit and ProveCommit is NI-PoRep (see for example discussion https://github.com/filecoin-project/FIPs/discussions/603 ). NI-PoRep allows to get a 100% storage cost saving and has other benefits (for example, removing the need of PreCommitDeposit) but has a ~8.5x proving overhead that we consider prohibitive for now. 
```

File: FIPS/fip-0059.md [hunk 4]
```diff
@@ -85,14 +94,15 @@ Will be included with implementation to be presented.
...
-In the new protocol, the SP first samples a set of `N_syn` positions and then proceeds to sample the 176 challenges from there. In order to be able to keep the same security as before, we need that the distribution of errors in the synthetic challenge set is as close as possible to the original distribution of errors in the layer. However the adversary can try different sets to get one where the fraction of wrongly encoded nodes is smaller than 3.9%. Say for example that adversary wants a fraction of 3.49% (this will allows it to pass with probability (1-0.0349)^176 > 2^{-9} > 2^{-10}),  we can show that if N = `N_syn` is large enough, then this is not possible. More in details, the probability that the number of wrongly encoded in the sythetic set is ≤ 0.0349*N is given by the bynomial probabilty:
+In the current PoRep protocol if 4% or more of nodes in a layer are wrongly encoded by the SP, then the ProveCommit step, where 176 prandom nodes are checked, will fail with large probability (larger than $1-2^{10}$).
```
Comment (arajasek) [inline FIPS/fip-0059.md:97]:
```suggestion
In the current PoRep protocol if 4% or more of nodes in a layer are wrongly encoded by the SP, then the ProveCommit step, where 176 random nodes are checked, will fail with large probability (larger than $1-2^{10}$).
```

(Unless "prandom" is a technical term I'm unaware of)

File: FIPS/fip-0059.md [hunk 5]
```diff
@@ -85,14 +94,15 @@ Will be included with implementation to be presented.
...
+In the new protocol, the SP first samples a set of `N_syn` positions and then proceeds to sample the 176 random nodes from there. In order to be able to keep the same security as before, we need that the distribution of errors in the synthetic challenge set is as close as possible to the original distribution of errors in the layer. However the adversary can try different sets to get one where the fraction of wrongly encoded nodes is smaller than 4%. Say for example that adversary wants to pass with probability $2^{-9}$, then he needs a set of synthetic challenges where at least a fraction $q$ are goods (correctly encoded) with $q$ = 0.9651 ($q^{176} = 2^{-9)}$).
+We can show that if `N_syn` is large enough, then this is not possible in practice. More in details: the probability that the number of correctly encoded nodes in the sythetic set is large or equal to $q$`N_syn` is given by the bynomial probabilty:
```
Comment (arajasek) [inline FIPS/fip-0059.md:99]:
```suggestion
We can show that if `N_syn` is large enough, then this is not possible in practice. More in details: the probability that the number of correctly encoded nodes in the sythetic set is large or equal to $q$`N_syn` is given by the binomial probabilty:
```

**Review by arajasek — APPROVED** [review]

File: FIPS/fip-0059.md [hunk 6]
```diff
@@ -53,27 +53,36 @@ Differences between currently deployed PoRep and Synthetic PoRep are limited to
...
+- No changes in `PreCommitSector` and `ProveCommitSector` methods;
+- `ProveCommitSector` passes the new prove type to the proof verification syscall;
 
 ### Proof changes
 
 - New parameters
-    - `N_syn` set to 2^18;
-    - `N_verified` set to 176 (same as *N_porep_challenges/k)* ;
+    - `N_syn` set to $2^{18}$;
+    - `N_verified` set to 176 (same as current *N_porep_challenges/k*) ;
```
Comment (Kubuxu) [inline FIPS/fip-0059.md:63]: What do you mean by `N_porep_challenges/k`? If the k is the partition count, then this is wrong.

We require 176 challenges, 180 are executed in practice in 10 partitions of 18 challenges.

File: FIPS/fip-0059.md [hunk 7]
```diff
@@ -85,14 +94,15 @@ Will be included with implementation to be presented.
...
+We can show that if `N_syn` is large enough, then this is not possible in practice. More in details: the probability that the number of correctly encoded nodes in the sythetic set is large or equal to $q$`N_syn` is given by the bynomial probabilty:
 
 $$
-P= \sum_{i=0}^{0.0349 N} \binom{N}{i}p^i (1-p)^{N-i} \text{ with } p ≥ 0.039
+P = \sum_{qN_{syn}}^{N_{syn}} \binom{N_{syn}}{i}p^i (1-p)^{N_{syn}-i} \text{ with } p ≤ 0.96
 $$
 
-and with N = `N_syn` ≥ 225000, then P < 2^{-80}. 
+and with `N_syn` = $2^18$, we have that $P < 2^{-138}$.
```
Comment (Kubuxu) [inline FIPS/fip-0059.md:105]:
```suggestion
and with `N_syn` = $2^{18}$, we have that $P < 2^{-138}$. 
```

**Comment by luckyparadise** [comment]
@arajasek @anorth could either of you merge this discussion, please? Thank you!

**Pull request closed without merging**